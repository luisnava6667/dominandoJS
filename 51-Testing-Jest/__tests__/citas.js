import Citas from "../js/classes/Citas";


describe('first Probar la clase de Citas', () => { 
    const citas = new Citas()
    const id =Date.now()
    test('should Agregar una nueva cita', () => {
      const citaObj = {
        id,
        mascota: 'Hook',
        propietario: 'Luis',
        telefono: '11111',
        fecha: '10/12/2014',
        hora: '10:00',
        sintomas: 'Solo Duerme'
      }
      
      citas.agregarCita(citaObj)
      expect(citas).toMatchSnapshot()
    })
    test('should Actualizar Cita|', () => {
      const citaActualizada = {
        id,
        mascota: 'Esperanzo',
        propietario: 'Luis',
        telefono: '11111',
        fecha: '10/12/2014',
        hora: '10:00',
        sintomas: 'Solo Duerme'
      }
      citas.editarCita(citaActualizada)
      expect(citas).toMatchSnapshot()
    })
    test('should Eliminar cita', () => {
      citas.eliminarCita(id)
      expect(citas).toMatchSnapshot()
    })
    
    
 })