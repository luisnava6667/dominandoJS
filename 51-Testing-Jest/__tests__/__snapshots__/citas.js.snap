// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first Probar la clase de Citas should Actualizar Cita| 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10/12/2014",
      "hora": "10:00",
      "id": 1701635314272,
      "mascota": "Esperanzo",
      "propietario": "Luis",
      "sintomas": "Solo Duerme",
      "telefono": "11111",
    },
  ],
}
`;

exports[`first Probar la clase de Citas should Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10/12/2014",
      "hora": "10:00",
      "id": 1701635314272,
      "mascota": "Hook",
      "propietario": "Luis",
      "sintomas": "Solo Duerme",
      "telefono": "11111",
    },
  ],
}
`;

exports[`first Probar la clase de Citas should Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
